src/pyats/contrib/plugins/topoup_plugin/topoup.py

```diff
@@ -0,0 +1,139 @@
+# Python
+import logging
+from time import sleep, time
+
+# pyAts
+from pyats.async_ import pcall
+from pyats.easypy.plugins.bases import BasePlugin
+
+# Logger
+log = logging.getLogger('ats.easypy.%s' % __name__)
+
+
+class TopologyUpPlugin(BasePlugin):
+    '''
+    Runs before job starts, to verify virtual topology is up and running
+    before executing the test script.
+    '''
+
+    @classmethod
+    def configure_parser(cls, parser, legacy_cli = True):
+        grp = parser.add_argument_group('TopologyUpPlugin')
+
+        if legacy_cli:
+            all_devices_up = ['-all_devices_up']
+            connection_check_timeout = ['-connection_check_timeout']
+            connection_check_interval = ['-connection_check_interval']
+        else:
+            all_devices_up = ['--all_devices_up']
+            connection_check_timeout = ['--connection_check_timeout']
+            connection_check_interval = ['--connection_check_interval']
+
+        # -all_devices_up
+        # --all_devices_up
+        grp.add_argument(*all_devices_up,
+                         dest='all_devices_up',
+                         action='store_true',
+                         default=False,
+                         help='Enable/Disable checking for topology up pre job execution')
+
+        # -connection_check_timeout
+        # --connection_check_timeout
+        grp.add_argument(*connection_check_timeout,
+                         dest='connection_check_timeout',
+                         action='store',
+                         default=120,
+                         help='Total time allowed for checking devices connectivity')
+
+        # -connection_check_interval
+        # --connection_check_interval
+        grp.add_argument(*connection_check_interval,
+                         dest='connection_check_interval',
+                         action='store',
+                         default=10,
+                         help='Time to sleep between device connectivity checks')
+
+        return grp
+
+
+    def pre_job(self, task):
+        '''Loop over all the topology devices and make sure they are up and running
+           before executing the test script.
+        '''
+
+        # Set the timers
+        start_time = time()
+        timeout = self.runtime.args.connection_check_timeout
+        interval = self.runtime.args.connection_check_interval
+
+        log.info("Connectivity check timeout is '{timeout}' and "
+            "connectivity check interval is '{interval}'".format(timeout=timeout, interval=interval))
+
+        # Looping over devices and make sure they are up
```

Comment:
comment should say that we are using parallel, not really looping

Reply:
Comment was from the old design, done.